### PR TITLE
chore(rxjs): update library to be compatable with rxjs and angular 6

### LINKED ISSUE #271: Angular 6, rxjs 6 support
Angular 6 is released, I think this will be a small fix about rxjs6. 

 > ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/observable/fromEvent' in '/Users/ismet.ozozturk/Desktop/mesele/mesele-frontend/node_modules/ngx-infinite-scroll/modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/observable/of' in '/Users/ismet.ozozturk/Desktop/mesele/mesele-frontend/node_modules/ngx-infinite-scroll/modules'

### Environment
- [ ] Version used: 0.8.4

### PR DESCRIPTION
**Pull request by deeg**
Closes #272
Fixes #271

**Summary**
This is a duplicate PR of #272 with all of the current comments addressed.

This PR fixes/implements the following **bugs/features**
- [X] Breaking changes

### Requirements
Make sure these boxes are checked:  
- [X] link to an issue
- [X] tests are updated (added/changed/removed)  
- [X] tests run successfully (local) 
- [X] tests run successfully (TravisCI) 
- [X] please follow this [styleguide](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#subject-line) for this pr title

### CODE REVIEW

**Comment by deeg** [comment]
@Roadirsh, @orizens I made all of the comments in the other PR to keep things moving.

Please let me know if you would like any other changes.

**Comment by deeg** [comment]
I have tested with my Angular 6, rxjs 6 app and everything is working.

You can try out in your app by replacing in your package.json:

```"ngx-infinite-scroll": "deeg/ngx-infinite-scroll#rxjs6-dist"```

File: src/modules/infinite-scroll.directive.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   Output,
   SimpleChanges
 } from '@angular/core';
-import { Subscription } from 'rxjs/Subscription';
+import {Subscription} from 'rxjs';
```
Comment (Charius) [inline src/modules/infinite-scroll.directive.ts]: I think around classname whitespaces should present

**Comment by orizens** [comment]
@deeg thanks for the effort!
will publish a beta for testing

**Pull request merged**